### PR TITLE
Tools#2 - Homework

### PR DESCRIPTION
**Pull request by mateusz-talma**

### CODE REVIEW

**Review by Kasperice — APPROVED** [review]
Changes look good

File: src/functions.cpp [hunk 1]
```diff
@@ -1,10 +1,9 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
-#include <algorithm>
+#include "functions.hpp"
 #include <iostream>
-#include <limits>
+#include <algorithm>
```
Comment (Marcinati) [inline src/functions.cpp:3]: First case: why do you remove limits header?
Second: sort headers.

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 src/*.cpp -I inc/ -o main -pedantic -Wall -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Wdisabled-optimization -Wformat=2 -Winit-self -Wlogical-op -Wmissing-declarations -Wmissing-include-dirs -Wnoexcept -Wold-style-cast -Woverloaded-virtual -Wredundant-decls -Wshadow -Wsign-conversion -Wsign-promo -Wstrict-null-sentinel -Wstrict-overflow=5 -Wswitch-default -Wundef -Werror -Wno-unused
```
Comment (Marcinati) [inline Makefile:2]: That's a lot of flag! Do you need all of them?

**Review by Marcinati — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (ziobron) [reply Makefile:2]: The more the better ;)

**Comment by ziobron** [comment]
+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }

File: Makefile (hunk 2 above)
Comment (mateusz-talma) [reply Makefile:2]: > That's a lot of flag! Do you need all of them?

Yes

**Pull request closed without merging**